products/views.py

```diff
@@ -1,3 +1,30 @@
-from django.shortcuts import render
+from django.views import View
+from django.http  import JsonResponse
 
-# Create your views here.
+from products.models import Product
+
+class ProductView(View):
+    def get(self, request, product_id):
+        if not Product.objects.filter(id=product_id).exists():
+            return JsonResponse({"message":"PRODUCT_DOES_NOT_EXIST"}, status=400)
+
+        product = Product.objects.get(id=product_id)
+
+        image_list = [image.image_url for image in product.detailimage_set.all()]
+
+        response = {
+            "id"               : product.id,
+            "name"             : product.name,
+            "price"            : int(product.price),
+            "style_code"       : product.style_code,
+            "origin"           : product.origin,
+            "manufacture_date" : product.manufacture_date,
+            "description"      : product.description,
+            "image_url"        : product.image_url,
+            "image_list"       : image_list,
+            "color"            : product.color.name,
+            "sub_category"     : product.sub_category.sub_category,
+            "category"         : product.sub_category.category.category,
+        }
```

Comment:
```suggestion
        product = Product.objects.get(id=product_id)
        
        response = {
            "id"               : product.id,
            "name"             : product.name,
            "price"            : int(product.price),
            "style_code"       : product.style_code,
            "origin"           : product.origin,
            "manufacture_date" : product.manufacture_date,
            "description"      : product.description,
            "image_url"        : product.image_url,
            "image_list"       :  [image.image_url for image in product.detailimage_set.all()],
            "color"            : product.color.name,
            "sub_category"     : product.sub_category.sub_category,
            "category"         : product.sub_category.category.category,
        }
```
product로부터 이미지 정보 가져오는 방법 잘 적용하셨습니다.!!!
image_list 변수가 1번만 사용되고, 해당 내용은 reponse에 있는 것이 더 가독성에 좋을 것 같습니다.
해당 내용만 수정하면 merge 하겠습니다.

Reply:
image_list 리스트 내포 식을 response 내부로 옮겼습니다! 또, 프론트와 상의하여 이중 Dict가 아니라 response를 그대로 응답하게끔 JsonResponse를 수정하였습니다.
<img width="851" alt="스크린샷 2021-08-09 오전 10 31 24" src="https://user-images.githubusercontent.com/84955774/128652548-0f3a21ea-639a-4fca-bd15-10b8ff1a196b.png">